services/meta/client.go

```diff
@@ -131,6 +133,54 @@ func (c *Client) Ping(checkAllMetaServers bool) error {
 	return fmt.Errorf(string(b))
 }
 
+// AcquireLease attempts to acquire the specified lease.
+// A lease is a logical concept that can be used by anything that needs to limit
+// execution to a single node.  E.g., the CQ service on all nodes may ask for
+// the "ContinuousQuery" lease. Only the node that acquires it will run CQs.
+// NOTE: Leases are not managed through the CP system and are not fully
+// consistent.  Any actions taken after acquiring a lease must be idempotent.
+func (c *Client) AcquireLease(name string) (*Lease, error) {
+	c.mu.RLock()
+	server := c.metaServers[0]
+	c.mu.RUnlock()
+	url := fmt.Sprintf("%s/lease?name=%s&nodeid=%d", c.url(server), name, c.nodeID)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+	case http.StatusConflict:
+		err = errors.New("another node owns the lease")
+	case http.StatusBadRequest:
+		b, e := ioutil.ReadAll(resp.Body)
+		if e != nil {
+			return nil, e
+		}
+		return nil, fmt.Errorf("meta service: %s", string(b))
+	case http.StatusInternalServerError:
```

Comment:
For these errors it should try against another meta-server instead of immediately returning back. The same type of thing that the exec method does.